packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -214,9 +214,7 @@ class Scaffold extends StatefulWidget {
     this.drawer,
     this.scrollableKey,
     this.appBarBehavior: AppBarBehavior.anchor
-  }) : super(key: key) {
-    assert((appBarBehavior == AppBarBehavior.scroll) ? scrollableKey != null : true);
```

Comment:
Yes it could however given the importance of the appBar and body parameters, it doesn't seem likely to ever be const-constructed.